Mathlib/Algebra/Polynomial/SumIteratedDerivative.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.BigOperators
+import Mathlib.Algebra.Polynomial.Degree.Lemmas
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Algebra.Polynomial.RingDivision
+
+/-!
+# Sum of iterated derivatives
+-/
+
+open Finset
+open scoped Nat
+
+namespace Polynomial
+
+variable {R : Type*}
+
+section Semiring
+
+variable [Semiring R]
+
+theorem sum_ideriv_apply_of_lt' {p : R[X]} {n : ℕ} (hn : p.natDegree < n) :
```

Comment:
I think that using `ideriv` as abrreviation for "iteratived derivative" is a new convention introduced in this PR, right?
But the rest of mathlib uses `iterate_derivative`. If it is only for these first two lemmas, I think we should just use the "long" version.

The def `sumIderiv` below seems fine to me.